x/gamm/pool-models/balancer/amm_joinpool_test.go

```diff
@@ -939,24 +977,51 @@ func TestCalcJoinSingleAssetTokensIn(t *testing.T) {
 					Weight: sdk.NewInt(100),
 				},
 			},
-			tokensIn:     sdk.NewCoins(sdk.NewInt64Coin("uosmo", 50_000), sdk.NewInt64Coin("uatom", 100_000)),
-			expectShares: sdk.NewInt(2_072_912_400_000_000 + 1_624_999_900_000),
+			tokensIn:             sdk.NewCoins(sdk.NewInt64Coin("uosmo", 50_000), sdk.NewInt64Coin("uatom", 100_000)),
+			expectShares:         sdk.NewInt(2_072_912_400_000_000 + 1_624_999_900_000), // ACTUAL: 2_074_537_403_302_024
+			expectToleranceError: false, 
+		},
+		{
+			// |a - b| / min(a, b) <= allowedErrRatio
+			// a = expected shares, b = actual shares
+			// 0.04000000047920000 = abs(2_599_999_968_750-2_499_999_968_800)/2_499_999_968_800
+			// 	Full solution: https://www.wolframalpha.com/input?i=abs%282599999968750-2499999968800%29%2F2499999968800
+			name: "Error Tolerance: One token in, expected > actual",
+			swapFee:              sdk.MustNewDecFromStr("0"),
+			poolAssets:           oneTrillionEvenPoolAssets,
+			tokensIn:             sdk.NewCoins(sdk.NewInt64Coin("uosmo", 50_000)),
+			expectShares:         sdk.NewInt(2_599_999_968_750),// ACTUAL: 2_499_999_968_800
```

Comment:
wdyt about adding comments for the calculations on why these are outside the range of balancer module's `allowedErrRatio`?

Reply:
I can do that